mauro-domain/src/test/groovy/org/maurodata/test/domain/datamodel/DataModelSpec.groovy

```diff
@@ -187,4 +187,50 @@ class DataModelSpec extends Specification {
         objectDiff.numberOfDiffs == 0
     }
 
+    def 'deep clone the datamodel -should deep copy object and all modelitems'() {
+        given:
+        DataModel original = testDataModel
+        when:
+        DataModel cloned = (DataModel) original.deepClone()
+        cloned.setAssociations()
+        then:
+        !cloned.is(original)
+
+        cloned.dataTypes == original.dataTypes
+        !cloned.dataTypes.is(original.dataTypes)  //groovy object equality. cloned is not original
+
+        cloned.allDataClasses == original.allDataClasses
+        !cloned.allDataClasses.is(original.allDataClasses)
+
+        cloned.dataElements == original.dataElements
+        !cloned.dataElements.is(original.dataElements)
+
+        cloned.enumerationValues == original.enumerationValues
+        !cloned.enumerationValues.is(original.enumerationValues)  //groovy object equal
+
+        cloned.allDataClasses.containsAll(cloned.dataElements.dataClass)
+        original.allDataClasses.containsAll(original.dataElements.dataClass)
+
+        DataClass originalDataClass1 = original.allDataClasses.find {it.label == 'My First DataClass'}
+        originalDataClass1.extendedBy.size() == 1
+
+        DataClass clonedDataClass1 = cloned.allDataClasses.find {it.label == 'My First DataClass'}
+        clonedDataClass1.extendedBy.size() == 1
+        DataClass clonedDataClass3 = cloned.allDataClasses.find {it.label == "My Third DataClass"}
+        clonedDataClass3.extendsDataClasses.size() == 1
+        clonedDataClass3.extendsDataClasses.first().label == 'My First DataClass'
+        clonedDataClass3.extendsDataClasses.first().is(clonedDataClass1)
+        clonedDataClass1.extendedBy.size() == 1
+        clonedDataClass1.extendedBy.first().is(clonedDataClass3)
+
+        cloned.dataElements.dataType.hashCode() != original.dataElements.dataType.hashCode()
+        cloned.dataElements.dataType.id == original.dataElements.dataType.id
+        cloned.dataTypes.containsAll(cloned.dataElements.dataType)
+        original.dataTypes.containsAll(original.dataElements.dataType)
+
+        !cloned.dataElements.dataType.is(original.dataElements.dataType)
+
+        ObjectDiff objectDiff = original.diff(cloned)
+        objectDiff.numberOfDiffs == 0
+    }
```

Comment:
This looks good, I think it would make sense to add a few key facets to the test which are used most:
* Metadata on Items, especially DataClasses and DataElements
* SummaryMetadata on Items, especially DataElements